powerscale/helper/synciq_policy_schema_datasource.go

```diff
@@ -0,0 +1,487 @@
+/*
+Copyright (c) 2023-2024 Dell Inc., or its subsidiaries. All Rights Reserved.
+
+Licensed under the Mozilla Public License Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://mozilla.org/MPL/2.0/
+
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package helper
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+// SynciqpolicyDatasourceSchema is a function that returns the schema for SyncIQPolicyDataSource.
+func SynciqpolicyDatasourceSchema(ctx context.Context) schema.Schema {
+	return schema.Schema{
+		Attributes: SyncIQPolicyDataSourceSchema(),
+	}
+}
+
+// SyncIQPolicyDataSourceSchema is a function that returns the schema for SyncIQPolicyDataSource.
+func SyncIQPolicyDataSourceSchema() map[string]schema.Attribute {
+	return map[string]schema.Attribute{
+		"id": schema.StringAttribute{
```

Comment:
Can we retrieve details about multiple SyncIQ policies?

Reply:
yes u can. If you do not specify id, all the policies would be fetched.

Reply:
Any reason we are not supporting list of ids?

Reply:
There is no API to filter by list of IDs, only by ID. If user wants to get multiple policies by their IDs, they can always put a loop on the datasource.